### DISCUSSION TITLE
Where are the arms handled in the project?

### DISCUSSION THREAD

**Discussion by RazielMartelus96** [DaXcess/LCVR#217] [General]
Hey there, im someone who tends to code a lot more minecraft plugins, but I do have an Aerospace Engineering Degree with a specialty in structural engineering, so wanted to see if I could have a wack at making the arms less likely to invert and look like ive seriously broken my arm haha. 

Which parts of the project are used to handle the arms?

**Comment by DaXcess** [comment]
Hey there, the code for the arms is mostly located in `VRPlayer.cs`
https://github.com/DaXcess/LCVR/blob/610e7ad6b02f0ef3486fc5853d74f7a67cab575e/Player/VRPlayer.cs#L238-L341

And a fixed version of the arms that are no longer broken is already ready on the development version
https://github.com/DaXcess/LCVR/blob/1bfa22f63a25bade62d1a8bc9b5e9d3af42db48f/Source/Player/VRPlayer.cs#L211-L351

The reason the arms looked broken was because apparently the game was using an IK chain constraint instead of a Two-bone constraint, even though the game had `X_arm_hint` in the hierarchy, making me believe the hints were actually used.

Turns out however, they weren't, so in the new version I replace the ChainIKConstraint for a TwoBoneIKContraint and that has fixed the arms since.
